pkg/apis/sources/v1beta2/ping_conversion_test.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"knative.dev/pkg/apis"
+)
+
+// implement apis.Convertible
+type dummyObject struct{}
+
+func (*dummyObject) ConvertTo(ctx context.Context, obj apis.Convertible) error {
+	return errors.New("Won't go")
+}
+
+func (*dummyObject) ConvertFrom(ctx context.Context, obj apis.Convertible) error {
+	return errors.New("Won't go")
+}
+
+func TestPingSourceConversionBadType(t *testing.T) {
+	good, bad := &PingSource{}, &dummyObject{}
+
+	if err := good.ConvertTo(context.Background(), bad); err == nil {
+		t.Errorf("ConvertTo() = %#v, wanted error", bad)
```

Comment:
The coding style preferred seems to be using `Error` instead of `Errorf`. Recent PRs are correcting the code in that direction.

Reply:
I think maybe you're referring to this PR: https://github.com/knative/eventing/pull/4178 , it was fixing things like:

```diff
- t.Errorf("ConvertTo() = %v", bad)
+ t.Error("ConvertTo() = ", bad)
```

thanks for the reminder, there're other places in this PR that need to be fixed.

Reply:
Fixed 2 occurrences in `pkg/apis/sources/v1beta2/ping_lifecycle_test.go`

Reply:
`Error` is only used when the variable is in the end or no variable. In the other PR `bad` is replaced with nil.